app/src/main/java/org/dslul/openboard/inputmethod/latin/spellcheck/AndroidWordLevelSpellCheckerSession.java

```diff
@@ -189,6 +226,7 @@ private static int getCheckabilityInScript(final String text, final int script)
      *  version of it "text" and the capitalized version of it "Text".
      */
     private boolean isInDictForAnyCapitalization(final String text, final int capitalizeType) {
+        updateLocale();
```

Comment:
I think maybe the getLocale() override and the mLocale update done in onCreate() could be enough to fix our issue. The locale is not likely to change during a Session lifecycle.

Reply:
That's what I implemented first, but if you switch your keyboard this wouldn't detect it—which I think it should because otherwise this would be somewhat useless.

Reply:
But I get that its not "clean" in any way to try to update the locale before every little part of the spellchecking process.